src/smartcontracts/codec/nothing.ts

```diff
@@ -0,0 +1,19 @@
+import { NothingValue } from "../typesystem";
+
+export class NothingCodec {
```

Comment:
What's the purpose of this class? Why do we need it? ( Or, to put it deeper: What's the purpose of nothing? :) )

Reply:
There are some types (in the ABI and / or in the Rust framework) encoded as nothing (no bytes) / require no decoding - e.g. the type `AsyncCall` (present both in the framework and in the ABIs),  `SCResult<()>` (this one only exists in the Rust framework).

`NothingType`, `NothingValue` have been added to handle these special types of the ABI.

CC: @andrei-marinica